check.py

```diff
@@ -53,16 +53,18 @@
                 sys.exit(1)
 
 # Update README.md flags section to match current library --help output
```

Comment:
If you move the output to a sphinx document, you could just use https://sphinxcontrib-programoutput.rtfd.io and https://sphinxcontrib-autoprogram.rtfd.io to stop rehardcoding this chunk.

Reply:
Wow nice.  I wanted something like this now that we have sphinx.  Totally, we can automate flags section of `README.md` too.

I was looking for a tool like this to automate example snippets within the doc-string.  My idea about docs is a little different than what you proposed above.  I think docs should be part of the code and not outside of it.   When writing docs, I want to include example snippets within class doc-string.  But problem is, those snippets may soon run out of sync with the actual code.  So all snippets must be within a directory, say `snippets` or we can re-use `examples` folder for it.  Then, I want to simply add something like this in the class doc-string

```
.. snippet::

   examples/snippets/acceptors.py#L10-L13
```

and this should render into actual code snippet either embedded via GitHub APIs or embedded during doc generation. 
 We can write tests for snippets too.  This will ensure that doc readers never run into surprises because I missed updating the doc somewhere :)

Unsure if linked plugins can do this out-of-box.  I'll dig into them.

Reply:
Including the program output is a separate task and should be implemented via those extensions.

As for the snippet inclusion, there's a few points.
1) doctests exist and they are testable. It's a well-known standard, use it. https://docs.python.org/3/library/doctest.html / https://www.sphinx-doc.org/en/master/usage/extensions/doctest.html. The runner for them is already added to the CI so just add those.
2) It is possible to include parts of files as code blocks in RST/Sphinx. You can use https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude which also allows you to include the contents partially and add some highlighting. Use this in the actual documents.
3) Do not use `.. literalinclude` in the docstrings. It's unreadable, untestable, and unmaintainable. It's a very bad idea.

You can include huge bits of docs in docstrings but many projects have moved away from this practice because a lot of prose may make it harder to deal with actually coding stuff. Plus you'll probably be unable to create https://diataxis.fr structured site (which you should). This is why these approaches are most often combined.